lib/zone-spec/fake-async-test.ts

```diff
@@ -0,0 +1,260 @@
+(function() {
+  interface ScheduledFunction {
+    endTime: number;
+    id: number,
+    func: Function;
+    args: any[];
+    delay: number;
+  }
+  
+  class Scheduler {
+    // Next scheduler id.
+    public nextId: number = 0;
+    
+    // Scheduler queue with the tuple of end time and callback function - sorted by end time.
+    private _schedulerQueue: ScheduledFunction[] = [];
+    // Current simulated time in millis.
+    private _currentTime: number = 0;
+    // Current scheduler function being processed.
+    private _current: ScheduledFunction = null;
+    
+    constructor() {}
+    
+    scheduleFunction(cb: Function, delay: number, args: any[] = [], id: number = -1) : number {
+      let currentId: number = id < 0 ? this.nextId++ : id;
+      let endTime = this._currentTime + delay;
+            
+      // Insert so that scheduler queue remains sorted by end time.
+      let newEntry: ScheduledFunction = {
+        endTime: endTime,
+        id: currentId,
+        func: cb,
+        args: args,
+        delay: delay
+      }
+      let i = 0;
+      for (; i < this._schedulerQueue.length; i++) {
+        let currentEntry = this._schedulerQueue[i];
+        if (newEntry.endTime < currentEntry.endTime) {
+          break;
+        }
+      }
+      this._schedulerQueue.splice(i, 0, newEntry);
+      return currentId;
+    }
+    
+    removeScheduledFunctionWithId(id: number): void {
+      for (let i = 0; i < this._schedulerQueue.length; i++) {
+        if (this._schedulerQueue[i].id == id) {
+          this._schedulerQueue.splice(i, 1);
+          break;
+        }
+      }
+    }
+    
+    tick(millis: number = 0): void {
+      this._currentTime += millis;
+      while (this._schedulerQueue.length > 0) {  
+        let current = this._schedulerQueue[0];        
+        if (this._currentTime < current.endTime) {
+          // Done processing the queue since it's sorted by endTime.
+          break;
+        } else {
+          // Time to run scheduled function. Remove it from the head of queue.
+          this._current = this._schedulerQueue.shift();
```

Comment:
Why store `_current` as a member on `this` instead of just keeping it a local variable?